src/rotten-tomatoes/index.js

```diff
@@ -1,3 +1,119 @@
-function rottenTomatoes(grid) {}
+//declaration and initialization of global variables
+const freshTomatoe = 1;
+const rottenTomatoe = 2;
+let timeToRotInMinute;
+let tomatoes = [];
 
+function rottenTomatoes(grid) 
+{
+    //check to ensure input is a valid parameter
+    if (!Array.isArray(grid) || grid.length === 0)
+    {
+        throw `Please enter a valid input`;
+    }
+
+    //record all the fresh tomatoes available in the grid
+    const totalFreshTomatoes = countFreshTomatoes(grid);
+    
+    //copy the input grid items to tomatoes 
+    tomatoes = grid.map(items => items);
```

Comment:
Why not initialize `tomatoes` here?

This doesn't do what you think it does, a shallow copy is made. Modifying `tomatoes` still mutates the original grid.

```js
    tomatoes = grid.map(items => [...items]);
```

That does a deep copy.

Reply:
- I wanted the `tomatoes` variable to be globally accessible, that is why I initialized it outside the function. 
- I have taken note of the shallow copy I made, to avoid mutation when such a case arises.

Sincerely, Sir I appreciate every effort you have made to review our code for best practice.  Thank you so much